src/analysis/special_functions/gamma.lean

```diff
@@ -519,3 +534,425 @@ end
 end complex
 
 end Gamma_has_deriv
+
+namespace real
+
+/-- The `Γ` function (of a real variable `s`). -/
+@[pp_nodot] def Gamma (s : ℝ) : ℝ := (complex.Gamma s).re
+
+lemma Gamma_eq_integral {s : ℝ} (hs : 0 < s) : Gamma s = ∫ x in Ioi 0, exp (-x) * x ^ (s - 1) :=
+begin
+  rw [Gamma, complex.Gamma_eq_integral (by rwa complex.of_real_re : 0 < complex.re s)],
+  dsimp only [complex.Gamma_integral],
+  simp_rw [←complex.of_real_one, ←complex.of_real_sub],
+  suffices : ∫ (x : ℝ) in Ioi 0, ↑(exp (-x)) * (x : ℂ) ^ ((s - 1 : ℝ) : ℂ) =
+    ∫ (x : ℝ) in Ioi 0, ((exp (-x) * x ^ (s - 1) : ℝ) : ℂ),
+  { rw [this, _root_.integral_of_real, complex.of_real_re], },
+  refine set_integral_congr measurable_set_Ioi (λ x hx, _),
+  push_cast,
+  rw complex.of_real_cpow (le_of_lt hx),
+  push_cast,
+end
+
+lemma Gamma_add_one {s : ℝ} (hs : s ≠ 0) : Gamma (s + 1) = s * Gamma s :=
+begin
+  simp_rw Gamma,
+  rw [complex.of_real_add, complex.of_real_one, complex.Gamma_add_one, complex.of_real_mul_re],
+  rwa complex.of_real_ne_zero,
+end
+
+theorem Gamma_nat_eq_factorial (n : ℕ) : Gamma (n + 1) = nat.factorial n :=
```

Comment:
Isn't it shorter to deduce it from the complex version? By the way, if you `open_locale nat`, you can use `n!` (here and elsewhere in the file).